File: inheritance.cpp
#include<iostream>

using namespace std;

class Shape {
 public:
    Shape(int w, int h) {
        width = w;
        height = h;
    }
    void draw() {
        cout << "Drawing a shape" << endl;
    }
 protected:
    int width;
    int height;
};

class Rectangle: public Shape {
 public:
    // call the constructor of the base class
    Rectangle(int w, int h): Shape(w, h) {
    }
    void draw() {
        Shape::draw(); // call the draw method of the base class
        cout << "Drawing a rectangle" << endl;
    }
};

int main() {
    Rectangle rect(5, 7);
    rect.draw();
    return 0;
}

File: inheritance.py
class Shape:
    def __init__(self, w, h):
        self.width = w
        self.height = h

    def draw(self):
        print("Drawing a shape")

class Rectangle(Shape):
    def __init__(self, w, h):
        super().__init__(w, h)  # call the constructor of the parent class

    def draw(self):
        super().draw()  # call the draw method of the parent class
        print("Drawing a rectangle")

rect = Rectangle(5, 7)
rect.draw()
